src/figuresClasses.ts

```diff
@@ -2,10 +1,75 @@
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c];
+    const maxSide = Math.max(...sides);
+    const sumOfTwoSides = sides.reduce((sum, side) => sum + side, 0) - maxSide;
+
+    if (sides.some((side) => side <= 0) || sumOfTwoSides <= maxSide) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('your error message');
```

Comment:
Consider providing a more descriptive error message here, such as 'Invalid radius: the radius must be greater than zero.'